### PR TITLE
Browse Page

### PR DESCRIPTION
**Pull request by RanDumSocks**
## Changes
* Added project browser page
  * Defaults (with no search parameters) to showing all projects
  * Tidied up naming schemes throughout

## Notes
Built off of PR #86

### CODE REVIEW

File: src/components/PreviewCompatibleImage.js [hunk 1]
```diff
@@ -16,8 +16,13 @@ const PreviewCompatibleImage = ({ imageInfo }) => {
     return <Img style={imageStyle} fluid={childImageSharp.fluid} alt={alt} />
   }
 
-  if (!!image && typeof image === 'string')
+  if (!!image && typeof image === 'string') {
     return <img style={imageStyle} src={image} alt={alt} />
+  }
+
+  if (typeof imageInfo === 'string') {
+    return <img style={imageStyle} src={imageInfo} alt={alt} />
```
Comment (chris-hammond) [inline src/components/PreviewCompatibleImage.js:24]: The fact that you need this has me worried. Makes me think that we're not processing images correctly for lunr (i.e. not tracking the correctly processed ones).

File: gatsby-node.js [hunk 2]
```diff
@@ -150,3 +180,18 @@ exports.createResolvers = ({ createResolvers }) => {
...
+
+exports.onPostBootstrap = () => {
+    var index = eJson('public/search_index.json');
+    var images = eJson('public/imageSrc.json').data.data.allStrapiProject.edges;
+
+    // en -> store
+    for (var i = 0; i < images.length; i++) {
+      var node = images[i].node;
+      var id = node.id;
+      var image = node.project_image.childImageSharp.fluid.src;
```
Comment (chris-hammond) [inline gatsby-node.js:192]: Ah-ha! Here's the problem. We need to save the _entire_ `childImageSharp` object, not just the `src` part, so we can pass the entire `sercSet` plus the width/aspectRatio/etc. to the gatsby-image plugin.

File: src/components/Navbar.js [hunk 3]
```diff
@@ -84,8 +84,8 @@ const Navbar = class extends React.Component {
             <Link className="navbar-item" to="/create-project">
               Create a Project
             </Link>
-            <Link className="navbar-item" to="/projects">
-              View All Projects
+            <Link className="navbar-item" to="/browse">
```
Comment (chris-hammond) [inline src/components/Navbar.js:87]: Huh. Can we just have this cool new functionality live at `/projects`. I don't love the `/browse` URI... You can delete the old projects.js page - we have it in git history if we need it...

File: src/components/Navbar.js [hunk 4]
```diff
@@ -84,8 +84,8 @@ const Navbar = class extends React.Component {
...
+              Browse Projects
```
Comment (chris-hammond) [inline src/components/Navbar.js:88]: Ditto for this - let's just keep calling the link "Projects" - the user will still certainly go there to browse.

File: src/components/PreviewCompatibleImage.js (hunk 1 above)
Comment (chris-hammond) [reply src/components/PreviewCompatibleImage.js:24]: For now, I think I'd like to remove this section, just to make sure we're passing the `childImageSharp` property correctly.

File: src/components/SearchField.js [hunk 5]
```diff
@@ -15,27 +15,30 @@ function SubmitTimeout() {
 }
 
 function Submit() {
-  navigate(`/project-search?keywords=${encodeURIComponent(queryText)}&page=1`)
+  navigate(`/browse?keywords=${encodeURIComponent(queryText)}&page=1`)
```
Comment (chris-hammond) [inline src/components/SearchField.js:18]: Ditto here for `/projects`

File: src/components/SearchPageNavigator.js [hunk 6]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import { navigate } from 'gatsby';
 
 function gotoPage(query, page) {
-  navigate(`/project-search?keywords=${encodeURIComponent(query)}&page=${page}`)
+  navigate(`/browse?keywords=${encodeURIComponent(query)}&page=${page}`)
```
Comment (chris-hammond) [inline src/components/SearchPageNavigator.js:5]: Ditto here for `/projects`

File: src/components/SearchResults.js [hunk 7]
```diff
@@ -4,51 +4,13 @@ import ProjectTile from '../components/ProjectTile.js';
...
-  }
-
   return(
     <section aria-live="assertive">
       <div className="columns is-multiline is-centered">
         {results.map(result => {
           return(
             <div key={result.id} className="column is-8-tablet is-6-desktop is-3-widescreen">
-              <Link to={result.id}><PreviewCompatibleImage imageInfo={result.imageData} /></Link>
+              <Link to={result.id}><PreviewCompatibleImage imageInfo={result.image} /></Link>
```
Comment (chris-hammond) [inline src/components/SearchResults.js:13]: Assuming you do `var image = node.project_image.childImageSharp` above, this code shouldn't have to change at all.

File: src/components/SearchField.js (hunk 5 above)
Comment (RanDumSocks) [reply src/components/SearchField.js:18]: Already fixed in PR #91

**Comment by RanDumSocks** [comment]
Fixed these requests in PR #91, which basically supersedes this PR completely. Should this one close and just focus on #91  ?

**Pull request closed without merging**